### PR TITLE
Feature/twitter data

### PR DESCRIPTION
**Pull request by gqinami**
Two major functionalities: 
- Connect via socket to receive twitter data (i.e. Mobs)
- Implement filter function (currently only for sightings, since predictions are still not available)

### CODE REVIEW

File: module/main.js [hunk 1]
```diff
@@ -31,9 +33,13 @@ require('../style.css');
             }
         }
 
+        var sightingsSinceDate = options.filter.sightingsSince;
+        var predictionsUntilDate = options.filter.predictionsUntil;
+        var pokemonIdsArray = options.filter.pokemonIds;
         var zoomLevel = options.zoomLevel;
         var timeRange = options.timeRange;
         var apiEndpoint = options.apiEndpoint;
+        var socketEndPoint = options.webSocketEndPoint;
```
Comment (johartl) [inline module/main.js]: The option is called `websocketEndpoint`

File: module/main.js [hunk 2]
```diff
@@ -242,25 +264,71 @@ require('../style.css');
...
-            if(route && route.removeFrom) {
+            if (route && route.removeFrom) {
 
                 route.removeFrom(mymap);
 
             }
 
         }
 
+        function filter(pokemonIds, sightingsSince, predictionsUntil) {
```
Comment (johartl) [inline module/main.js]: The filter function receives only one parameter which is an object containing the three filter criteria:

```
function filter(filterOptions) {
  let pokemonIds = filterOptions.pokemonIds;
  let sightingsSince = filterOptions.sightingsSince;
  let predictionsUntil = filterOptions.predictionsUntil;
  ...
}
```

File: script.js [hunk 3]
```diff
@@ -1,7 +1,11 @@
 var htmlElement = 'mapid';
 var options = {
 
+    filter : {
+        pokemonIds: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10], sightingsSince: new Date(2016, 5, 20), predictionsUntil: 0
+    },
     apiEndpoint: 'http://pokedata.c4e3f8c7.svc.dockerapp.io:65014',
+    webSocketEndPoint: 'pokedata.c4e3f8c7.svc.dockerapp.io:65024',
```
Comment (johartl) [inline script.js]: Again `websocketEndpoint`

File: module/main.js [hunk 4]
```diff
@@ -33,13 +33,10 @@ var DataService = require('./DataService.js');
...
         }
 
-        var sightingsSinceDate = options.filter.sightingsSince;
-        var predictionsUntilDate = options.filter.predictionsUntil;
-        var pokemonIdsArray = options.filter.pokemonIds;
         var zoomLevel = options.zoomLevel;
         var timeRange = options.timeRange;
         var apiEndpoint = options.apiEndpoint;
-        var socketEndPoint = options.webSocketEndPoint;
+        var socketEndPoint = options.websocketEndPoint;
```
Comment (johartl) [inline module/main.js]: Sorry but it's `websocketEndpoint` (no capital P)

File: script.js [hunk 5]
```diff
@@ -2,10 +2,10 @@ var htmlElement = 'mapid';
 var options = {
 
     filter : {
-        pokemonIds: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10], sightingsSince: new Date(2016, 5, 20), predictionsUntil: 0
+        pokemonIds: null, sightingsSince: new Date(2015, 5, 20), predictionsUntil: 0
     },
     apiEndpoint: 'http://pokedata.c4e3f8c7.svc.dockerapp.io:65014',
-    webSocketEndPoint: 'pokedata.c4e3f8c7.svc.dockerapp.io:65024',
+    websocketEndPoint: 'pokedata.c4e3f8c7.svc.dockerapp.io:65024',
```
Comment (johartl) [inline script.js]: Same here

File: module/main.js [hunk 6]
```diff
@@ -283,7 +280,11 @@ var DataService = require('./DataService.js');
 
         }
 
-        function filter(pokemonIds, sightingsSince, predictionsUntil) {
+        function filter(filterOptions) {
+
+            var sightingsSince = options.filter.sightingsSince;
+            var predictionsUntil = options.filter.predictionsUntil;
+            var pokemonIds = options.filter.pokemonIds;
```
Comment (johartl) [inline module/main.js]: I think this should be

```
var sightingsSince = filterOptions.sightingsSince;
var predictionsUntil = filterOptions.predictionsUntil;
var pokemonIds = filterOptions.pokemonIds;
```

Comment (gqinami) [reply module/main.js]: yeah you are right. Thanks for the comments

**Comment by johartl** [comment]
Seems fine now

**Pull request merged**